src/tree.rs

```diff
@@ -1126,7 +1126,11 @@ fn detect_cycles(parents: &[ResolvedParent]) -> Option<Index> {
         let mut grandparent_index = parent.index().and_then(|index| parents[index].index());
         while let (Some(i), Some(j)) = (parent_index, grandparent_index) {
             if i == j {
-                return Some(i);
+                println!("CYCLE INDEX: {}", i);
+                println!("DC BUILDER: {:#?}", builder);
+                break_cycles(builder, i).expect("Can't break cycles");
+                // return Some(i);
+                return None;
```

Comment:
Sanity check: I'm assuming `detect_cycles` should no longer return Some(i), let me know if I'm lying.